### PR TITLE
Call notifyChanges in updateRow instead of detectChanges

### LINKED ISSUE #5934: 8.2.* IgxGridComponent.updateRow poor performance
## Description  
After updating from 8.1.5 to 8.2.2, the performance of updateRow has degraded significantly , even if newer features are disabled such as advanced filtering.

 * igniteui-angular version: 8.2.2
 * browser: Chrome Version 73.0.3683.103 (Official Build) (32-bit)

## Steps to reproduce  
Click on the below links to view performance difference.

1. Create igx-grid with realtime updating rows via the updateRow function call

## Result  
The grid updates very slowly, with greatly increased time needed for each updateRow call.

## Expected result  
The grid should perform as it did in version 8.1.5

## Attachments  
igniteui-angular@8.1.5:  (grid updates normally)
    https://stackblitz.com/edit/angular-wkbmwm

igniteui-angular@8.2.2: (grid updates slowly)
    https://stackblitz.com/edit/angular-wkbmwm-vqzsup

### PR DESCRIPTION
**Pull request by wnvko**
We were calling `detectChanges` on each call to grid's `updateRow`. This made the grid unresponsive if `updateRow` was called very often. With this fix the grid will update when necessary without make the grid unresponsive.

Closes #5934 

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/grid-base.component.ts [hunk 1]
```diff
@@ -3947,7 +3947,13 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
...
             const row = new IgxRow(rowSelector, -1, this.gridAPI.getRowData(rowSelector));
             this.gridAPI.update_row(row, value);
-            this.cdr.detectChanges();
+
+            // TODO: fix for #5934 and probably break for #5763
+            // consider adding of third optional boolean parameter in updateRow.
+            // If developer set this parameter to true we should call notifyChanges(true), and
+            // vise-versa if developer set it to false we should call notifyChanges(false).
+            // The parameter should default to false
+            this.notifyChanges();
```
Comment (demingjs) [inline projects/igniteui-angular/src/lib/grids/grid-base.component.ts]: Would UpdateCell benefit from this change as well?  I haven't tested specifically with that but see the detectChanges is being used there as well.

Comment (wnvko) [reply projects/igniteui-angular/src/lib/grids/grid-base.component.ts]: @demingjs 
No, definitely no. This will break issue #5763.
There is a case when you may need to update a cell after another cell is changed, e.g. you have row with price, quantity and total. If user changes the value of let say quantity you would want to change the value of total by calling `updateCell`. If we do not call `detectChanges` at the end of `updateCell` total cell will not display updated value.
To summarize - if you need to update quickly the values of many cell, as in your case, you should call `updateRow`. If you need to change single cell you should call `updateCell`.
